diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -417,6 +417,19 @@ namespace diff_drive_controller
     controller_nh.param("angular/z/max_jerk"               , limiter_ang_.max_jerk               ,  limiter_ang_.max_jerk              );
     controller_nh.param("angular/z/min_jerk"               , limiter_ang_.min_jerk               , -limiter_ang_.max_jerk              );
 
+    // Limits linear velocity and acceleration along body frame X axis of each individual wheel.
+    controller_nh.param("wheel/x/has_velocity_limits"    , limiter_wheel_.has_velocity_limits    , limiter_wheel_.has_velocity_limits    );
```

Comment:
Specifying the x-axis here feels really awkward— is there a possible refactoring of the `SpeedLimiter` class which would permit that not to be necessary?

Reply:
From what I can see, the relation to the X-axis only affects the parameter topic, doesn't seem to have any affect on usage in code.

Reply:
The wheels are joints, so instead of specifying parameters we should be reading the limits from the joint interfaces.
Indeed, the parameters for the linear and angular limits are inspired and some parameters that can't be specified in the URDF for the joints, like the accelerations (at least by the time we implemented this, IIRC).

Reply:
https://github.com/ros-controls/ros_control/tree/kinetic-devel/joint_limits_interface/include/joint_limits_interface

Look at `joint_limits_urdf.h` and `joint_limits_rosparam.h`.
Not that you won't get accelerations from the URDF (not supported), so for that the rosparam is needed. I'm not completely sure, but I think you simply need to chain them: urdf first, rosparam after. This way rosparam will set acceleration and jerk, and also overwrite the urdf values if that's ever needed.